pkg/deploy/error.go

```diff
@@ -18,3 +20,19 @@ func isDeploymentNotFoundError(err error) bool {
 	}
 	return false
 }
+
+func isOperationPreemptedError(err error) bool {
```

Comment:
Are `requestErr`  the same as `serviceErr`? Do we want to make some distinction here? Would the difference between them warrant any different action? If not, looks good as is.